.github/workflows/secureSignal-cd.yaml

```diff
@@ -37,22 +45,30 @@ jobs:
         run: npm install
       - name: Build
         run: npm run build:esp -- --mode=${{ matrix.target }}
-      - uses: actions/upload-artifact@v2
+      - uses: actions/upload-artifact@v4
         with:
           name: ${{ matrix.target }}SecureSignalScript
           path: ./dist/uid2SecureSignal.js
+  
   deployment:
     needs: [build]
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    
     strategy:
       matrix:
         environment: [Integ, production]
```

Comment:
Should Integ be lower case? So that it matches development and production

Reply:
You can't set type on action inputs - who knows why

Reply:
Updated